### ISSUE TITLE
Ruletile error

### ISSUE THREAD

**Issue by emilpedersen** [Unity-Technologies/2d-extras#14]
I get this error on iOS works fine in Unity Editor.

```
LevelNew:TileFrom(TileType)
LevelNew:AddTile(Vector3)
LevelNew:Update()
 
(Filename: /Users/builduser/buildslave/unity/build/artifacts/generated/common/runtime/DebugBindings.gen.cpp Line: 51)

NullReferenceException: A null value was found where an object instance was required.
  at UnityEngine.Tilemaps.TileBase.RefreshTile (Vector3Int position, UnityEngine.Tilemaps.ITilemap tilemap) [0x00000] in <filename unknown>:0 
  at UnityEngine.Tilemaps.Tilemap.SetTile (Vector3Int position, UnityEngine.Tilemaps.TileBase tile) [0x00000] in <filename unknown>:0 
  at LevelNew.Update () [0x00000] in <filename unknown>:0 
UnityEngine.Tilemaps.Tilemap:SetTile(Vector3Int, TileBase)
LevelNew:Update()
 
(Filename: currently not available on il2cpp Line: -1)

Error running RefreshTile for previous TileNullReferenceException: A null value was found where an object instance was required.
  at UnityEngine.Tilemaps.TileBase.RefreshTile (Vector3Int position, UnityEngine.Tilemaps.ITilemap tilemap) [0x00000] in <filename unknown>:0 
  at UnityEngine.Tilemaps.Tilemap.SetTile (Vector3Int position, UnityEngine.Tilemaps.TileBase tile) [0x00000] in <filename unknown>:0 
  at LevelNew.Update () [0x00000] in <filename unknown>:0 
UnityEngine.Tilemaps.Tilemap:SetTile(Vector3Int, TileBase)
LevelNew:Update()
 
(Filename: currently not available on il2cpp Line: -1)

NullReferenceException: A null value was found where an object instance was required.
  at UnityEngine.Tilemaps.TileBase.RefreshTile (Vector3Int position, UnityEngine.Tilemaps.ITilemap tilemap) [0x00000] in <filename unknown>:0 
  at UnityEngine.RuleTile.RefreshTile (Vector3Int location, UnityEngine.Tilemaps.ITilemap tileMap) [0x00000] in <filename unknown>:0 
  at UnityEngine.Tilemaps.Tilemap.SetTile (Vector3Int position, UnityEngine.Tilemaps.TileBase tile) [0x00000] in <filename unknown>:0 
  at LevelNew.Update () [0x00000] in <filename unknown>:0 
UnityEngine.Tilemaps.Tilemap:SetTile(Vector3Int, TileBase)
LevelNew:Update()
 
(Filename: currently not available on il2cpp Line: -1)

Error running RefreshTile for previous TileNullReferenceException: A null value was found where an object instance was required.
  at UnityEngine.Tilemaps.TileBase.RefreshTile (Vector3Int position, UnityEngine.Tilemaps.ITilemap tilemap) [0x00000] in <filename unknown>:0 
  at UnityEngine.RuleTile.RefreshTile (Vector3Int location, UnityEngine.Tilemaps.ITilemap tileMap) [0x00000] in <filename unknown>:0 
  at UnityEngine.Tilemaps.Tilemap.SetTile (Vector3Int position, UnityEngine.Tilemaps.TileBase tile) [0x00000] in <filename unknown>:0 
  at LevelNew.Update () [0x00000] in <filename unknown>:0 
UnityEngine.Tilemaps.Tilemap:SetTile(Vector3Int, TileBase)
LevelNew:Update()
 
(Filename: currently not available on il2cpp Line: -1)
```

**Comment by ChuanXin-Unity** [comment]
Will check this out!

**Comment by AliBenba** [comment]
I have got the same error. Any workaround ? Any fix soon ?

**Comment by SungJJinKang** [comment]
please fix this error, i got this error too on ios

**Comment by Jhsosa** [comment]
Any news on this? Seems to be an issue when setting any tile in iOS via script.

**Comment by denisbondare** [comment]
Same thing on Android and WebGL (Unity 2017.3.0f3). Any news? How can I workaround it?

**Comment by ChuanXin-Unity** [comment]
Sorry, this is fixed in Unity 2017.4.1f1.
